op-challenger/game/fault/contracts/oracle_test.go

```diff
@@ -105,12 +105,45 @@ func TestPreimageOracleContract_Squeeze(t *testing.T) {
 }
 
 func TestGetActivePreimages(t *testing.T) {
-	stubRpc, oracle := setupPreimageOracleTest(t)
 	blockHash := common.Hash{0xaa}
+	_, oracle, proposals := setupPreimageOracleTestWithProposals(t, batching.BlockByHash(blockHash))
+	preimages, err := oracle.GetActivePreimages(context.Background(), blockHash)
+	require.NoError(t, err)
+	require.Equal(t, proposals, preimages)
+}
+
+func TestGetProposalMetadata(t *testing.T) {
+	blockHash := common.Hash{0xaa}
+	block := batching.BlockByHash(blockHash)
+	stubRpc, oracle, proposals := setupPreimageOracleTestWithProposals(t, block)
+	preimages, err := oracle.GetProposalMetadata(context.Background(), block, []gameTypes.LargePreimageIdent{
+		proposals[0].LargePreimageIdent,
+		proposals[1].LargePreimageIdent,
+		proposals[2].LargePreimageIdent,
+	})
+	require.NoError(t, err)
+	require.Equal(t, proposals, preimages)
+
+	// Fetching a proposal that doesn't exist should return an empty metadata object.
+	ident := gameTypes.LargePreimageIdent{Claimant: common.Address{0x12}, UUID: big.NewInt(123)}
```

Comment:
This isn't the behaviour I would have expected from the actual contract - I'd have thought the call would fail and return an error. Could be wrong on that but we probably should do a manual test to check this - it will likely turn up in e2e tests when we have them but could take us a while to identify this as the cause by then.

You can probably just hack something up locally really quick to confirm the actual behaviour - I don't think we need a test we can commit for this since e2e will cover it naturally.